src/components/Settings/SettingsMain.tsx

```diff
@@ -85,6 +99,18 @@ const SettingsMain: React.FC = () => {
         intl.formatMessage(messages.validationApplicationUrlTrailingSlash),
         (value) => !value || !value.endsWith('/')
       ),
+    proxyHostname: Yup.string().when('proxyEnabled', {
+      is: true,
+      then: Yup.string().required(
+        intl.formatMessage(messages.proxyValidationHostname)
+      ),
+    }),
+    proxyPort: Yup.string().when('proxyEnabled', {
+      is: true,
+      then: Yup.string().required(
+        intl.formatMessage(messages.proxyValidationPort)
+      ),
+    }),
```

Comment:
Should be validated as number.  See https://github.com/sct/overseerr/blob/ca184728e912c94e40484a55edfab01cae696a1b/src/components/Settings/SettingsPlex.tsx#L162-L171 for an example.